File: engage/javascript/packages/engage-sdk/src/entities/movieEntity.ts
import {
  ContentAvailability,
  PlatformSpecificUri,
  RatingSystem,
} from "../common";
import { InternalEntity } from "./entity";
import { VideoEntity } from "./videoEntity";

export interface MovieEntity extends VideoEntity {
  playbackUri: string;
  infoPageUri: string;
  releaseDateEpochMillis: number;
  contentAvailability: ContentAvailability;
  durationMillis: number;
  genres: string[];
  isDownloadedOnDevice: boolean;
  contentRatings: RatingSystem[];
  platformSpecificPlaybackUris: PlatformSpecificUri[];
}

interface InternalMovieEntity extends MovieEntity, InternalEntity {
  type: "com.google.android.engage.js.converter.dto.entities.MovieEntity";
}

export function buildMovieEntity(
  movieEntity: MovieEntity
): InternalMovieEntity {
  return {
    ...movieEntity,
    type: "com.google.android.engage.js.converter.dto.entities.MovieEntity",
  };
}

File: engage/kotlin/converter/src/main/kotlin/com/google/android/engage/js/converter/dto/entities/MovieEntity.kt
package com.google.android.engage.js.converter.dto.entities

import com.google.android.engage.js.converter.dto.common.AvailabilityTimeWindow
import com.google.android.engage.js.converter.dto.common.ContentAvailability
import com.google.android.engage.js.converter.dto.common.Image
import com.google.android.engage.js.converter.dto.common.PlatformSpecificUri
import com.google.android.engage.js.converter.dto.common.RatingSystem
import com.google.android.engage.js.converter.dto.common.WatchNextType
import kotlinx.serialization.Serializable

@Serializable
data class MovieEntity(
    val posterImages: List<Image>,
    val entityId: String,
    val name: String,
    val lastEngagementTimeMillis: Long,
    val watchNextType: WatchNextType,
    val lastPlaybackPositionTimeMillis: Long,
    val availabilityTimeWindows: List<AvailabilityTimeWindow>,

    val playbackUri: String,
    val infoPageUri: String,
    val releaseDateEpochMillis: Long,
    val contentAvailability: ContentAvailability,
    val durationMillis: Long,
    val genres: List<String>,
    val contentRatingsLegacy: List<String>,
    val isDownloadedOnDevice: Boolean,
    val contentRatings: List<RatingSystem>,
    val platformSpecificPlaybackUris: List<PlatformSpecificUri>,
): Entity()
